jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala

```diff
@@ -84,19 +87,39 @@ object JmsTextMessage {
    * Java API: create  [[JmsTextMessage]]
    */
   def create(body: String, headers: util.Set[JmsHeader]) =
-    JmsTextMessage(body = body, headers = headers.toSet, properties = Map.empty)
+    JmsTextMessage(body = body, headers = headers.asScala.toSet, properties = Map.empty)
 
   /**
    * Java API: create  [[JmsTextMessage]]
    */
   def create(body: String, properties: util.Map[String, Any]) =
-    JmsTextMessage(body = body, headers = Set.empty, properties = properties.toMap)
+    JmsTextMessage(body = body, headers = Set.empty, properties = properties.asScala.toMap)
 
   /**
    * Java API: create  [[JmsTextMessage]]
    */
   def create(body: String, headers: util.Set[JmsHeader], properties: util.Map[String, Any]) =
-    JmsTextMessage(body = body, headers = headers.toSet, properties = properties.toMap)
+    JmsTextMessage(body = body, headers = headers.asScala.toSet, properties = properties.asScala.toMap)
+}
+
+trait AcknowledgeMode {
+  val jmsAcknowledgeMode: Int
```

Comment:
I think using a `final class AcknowledgeMode(mode: Int)` and providing constants in the companion for the common modes would be cleaner than creating a type hierarchy just to provide different values (and will still allow custom ones as requested).